File: tests/js/ecma_operators/comparison_operators.js
// vybe-test: js/ecma_operators/comparison_operators
// origin: languages/js/tests/js/test_ecma_operators.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

__check(__line(1 < 2), "true");
__check(__line(2 > 1), "true");
__check(__line(1 <= 1), "true");
__check(__line(1 >= 1), "true");

File: tests/fortran/control_flow/comparison_operators.f90
! vybe-test: fortran/control_flow/comparison_operators
! origin: languages/fortran/tests/fortran/test_control_flow.rs

program test
    if (1 < 2) print *, "lt"
    if (2 > 1) print *, "gt"
    if (1 <= 1) print *, "le"
    if (1 >= 1) print *, "ge"
    if (1 == 1) print *, "eq"
    if (1 /= 2) print *, "ne"
end program test
